src/read_agent_logging.in

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+# Read metrics from the Stackdriver Logging agent. This expects the Prometheus
+# text format. It currently only supports what's needed for the Logging agent.
+#
+# This script handles the input line by line, and keeps running until EOF. This
+# is the recommended behavior for long-running collectd plugins, to reduce
+# start-up cost.
+
+# Read first arg, with a safe default for testing.
+URL="${1:-http://localhost:24231/metrics}"
+
+# Default to 10s if unset, as a fail-safe mechanism.
+INTERVAL=${COLLECTD_INTERVAL:-10}
+
+# The part of the input that sed looks at looks like:
+#   stackdriver_successful_requests_count{grpc="false"} 255
+#   stackdriver_failed_requests_count{grpc="false",code="400"} 14
+#   stackdriver_failed_requests_count{grpc="false",code="404"} 18
+#   stackdriver_failed_requests_count{grpc="false",code="503"} 8
+#   stackdriver_ingested_entries_count 7050
+#   stackdriver_dropped_entries_count 25
+#
+# The expected output follows:
+#   PUTVAL /agent-request_count/derive-200 N:255
+#   PUTVAL /agent-request_count/derive-400 N:14
+#   PUTVAL /agent-request_count/derive-404 N:18
+#   PUTVAL /agent-request_count/derive-503 N:8
+#   PUTVAL /agent-log_entry_count/derive-200 N:7050
+#   PUTVAL /agent-entries_retry_count/derive-200 N:73
+#   PUTVAL /agent-log_entry_count/derive-500 N:25
```

Comment:
Nit: dropped entries are not the result of 500s -- those would be retried.

Reply:
Changed to 418, because we don't really know the reason, whether it's client or server side. This should be fixed once we have partial error support in the Logging agent.

Reply:
Perfect.